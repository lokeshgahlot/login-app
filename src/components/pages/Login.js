import React, {Component} from "react";
import PropTypes from "prop-types";
import {Redirect, Link} from "react-router-dom";
import { Button, FormGroup, FormControl, ControlLabel, ButtonToolbar} from "react-bootstrap";
import WithInputValidator from "../WithInputValidator";
import {connect} from "react-redux";
import {authentication} from "../../actions/authentication.actions";
import "../../scss/form.scss";

export class LoginPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      alertMsg: "",
      redirect: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  validateForm() {
    return this.state.email.length > 0 && this.state.password.length > 0;
  }

  handleChange(event) {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  handleSubmit(event) {
    const {authentication} = this.props;
    const {email, password} = this.state;
    authentication({email, password});
    event.preventDefault();
  }

  render() {
    const {isAuthenticated} = this.props;
    if (isAuthenticated) {
      return <Redirect to={{ pathname: "/greeting", state: { from: this.props.location }}}/>;
    }

    return (
      <div className="login top-buffer">
        <form className="form" onSubmit={this.handleSubmit}>
          <FormGroup controlId="email" bsSize="large">
            <ControlLabel>Email</ControlLabel>
            <WithInputValidator>
              <FormControl
                autoFocus
                type="email"
                required
                value={this.state.email}
                onChange={this.handleChange}
              /> 
            </WithInputValidator>
          </FormGroup>
          <FormGroup controlId="password" bsSize="large">
            <ControlLabel>Password</ControlLabel>
            <FormControl
              value={this.state.password}
              onChange={this.handleChange}
              type="password"
            />
          </FormGroup>
          
          <ButtonToolbar className="pull-right">
            <Link to={{pathname: "/sign-up"}}>
              <Button bsSize="large"> Register </Button>
            </Link>
            <Button
              bsSize="large"
              bsStyle="primary"
              disabled={!this.validateForm()}
              type="submit"
            >
              Login
            </Button>
          </ ButtonToolbar>
        </form>
      </div>
    );
  }
}

LoginPage.propTypes = {
  isAuthenticated: PropTypes.bool,
  authentication: PropTypes.func
};

export const mapSTP = (state) => ({
  isAuthenticated: state.get("authenticationInfo").get("isAuthenticated")
});

export default connect(mapSTP, {
  authentication
})(LoginPage);