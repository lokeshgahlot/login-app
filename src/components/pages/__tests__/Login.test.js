import React from "react";
import {shallow} from "enzyme";
import {LoginPage, mapSTP} from "../Login";
import {fromJS} from "immutable";

describe("<LoginPage />", () => { 
  it ("renders correctly", () => {
    const wrapper = shallow(<LoginPage isAuthenticated={false}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it ("redirects", () => {
    const wrapper = shallow(<LoginPage isAuthenticated={true}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it ("calls authentication on form submit", () => {
    const fn = jest.fn();
    const wrapper = shallow(<LoginPage isAuthenticated = {false} authentication={fn}/>);
    const instance = wrapper.instance();
    const expectedResult = {
      email: "test@test.com", 
      password: "test"
    };
    instance.setState(expectedResult);
    wrapper.find(".form").simulate("submit", {preventDefault: () => {}});
    expect(fn.mock.calls.length).toBe(1);
    expect(fn.mock.calls[0][0]).toEqual(expectedResult);
  });

  it ("mapSTP()", () => {
    const state = fromJS({
      authenticationInfo: {
        isAuthenticated: true
      }
    });
    expect(mapSTP(state)).toEqual({isAuthenticated: true});
  });

});