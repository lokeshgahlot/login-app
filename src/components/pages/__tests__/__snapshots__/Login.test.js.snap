// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> redirects 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage
    isAuthenticated={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "push": false,
      "to": Object {
        "pathname": "/greeting",
        "state": Object {
          "from": undefined,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "push": false,
        "to": Object {
          "pathname": "/greeting",
          "state": Object {
            "from": undefined,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<LoginPage /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage
    isAuthenticated={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="form"
        onSubmit={[Function]}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="large"
          controlId="email"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Email
          </ControlLabel>
          <WithInputVaildator
            message="Invalid email"
            pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
            type="email"
          >
            <FormControl
              autoFocus={true}
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </WithInputVaildator>
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          bsSize="large"
          controlId="password"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="password"
            value=""
          />
        </FormGroup>
        <ButtonToolbar
          bsClass="btn-toolbar"
          className="pull-right"
        >
          <Link
            replace={false}
            to={
              Object {
                "pathname": "/sign-up",
              }
            }
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="large"
              bsStyle="default"
              disabled={false}
            >
               Register 
            </Button>
          </Link>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="primary"
            disabled={true}
            type="submit"
          >
            Login
          </Button>
        </ButtonToolbar>
      </form>,
      "className": "login top-buffer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="email"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Email
            </ControlLabel>
            <WithInputVaildator
              message="Invalid email"
              pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
              type="email"
            >
              <FormControl
                autoFocus={true}
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                required={true}
                type="email"
                value=""
              />
            </WithInputVaildator>
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="password"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Password
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormGroup>,
          <ButtonToolbar
            bsClass="btn-toolbar"
            className="pull-right"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/sign-up",
                }
              }
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="large"
                bsStyle="default"
                disabled={false}
              >
                 Register 
              </Button>
            </Link>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="large"
              bsStyle="primary"
              disabled={true}
              type="submit"
            >
              Login
            </Button>
          </ButtonToolbar>,
        ],
        "className": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "bsSize": "large",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Email
              </ControlLabel>,
              <WithInputVaildator
                message="Invalid email"
                pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
                type="email"
              >
                <FormControl
                  autoFocus={true}
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  required={true}
                  type="email"
                  value=""
                />
              </WithInputVaildator>,
            ],
            "controlId": "email",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Email",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Email",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <FormControl
                  autoFocus={true}
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  required={true}
                  type="email"
                  value=""
                />,
                "message": "Invalid email",
                "pattern": "^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$",
                "type": "email",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": true,
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "onChange": [Function],
                  "required": true,
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "bsSize": "large",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Password
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="password"
                value=""
              />,
            ],
            "controlId": "password",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Password",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Password",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "btn-toolbar",
            "children": Array [
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/sign-up",
                  }
                }
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="large"
                  bsStyle="default"
                  disabled={false}
                >
                   Register 
                </Button>
              </Link>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="large"
                bsStyle="primary"
                disabled={true}
                type="submit"
              >
                Login
              </Button>,
            ],
            "className": "pull-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="large"
                  bsStyle="default"
                  disabled={false}
                >
                   Register 
                </Button>,
                "replace": false,
                "to": Object {
                  "pathname": "/sign-up",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsSize": "large",
                  "bsStyle": "default",
                  "children": " Register ",
                  "disabled": false,
                },
                "ref": null,
                "rendered": " Register ",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsSize": "large",
                "bsStyle": "primary",
                "children": "Login",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="form"
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="email"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Email
            </ControlLabel>
            <WithInputVaildator
              message="Invalid email"
              pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
              type="email"
            >
              <FormControl
                autoFocus={true}
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                required={true}
                type="email"
                value=""
              />
            </WithInputVaildator>
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="password"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Password
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              type="password"
              value=""
            />
          </FormGroup>
          <ButtonToolbar
            bsClass="btn-toolbar"
            className="pull-right"
          >
            <Link
              replace={false}
              to={
                Object {
                  "pathname": "/sign-up",
                }
              }
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="large"
                bsStyle="default"
                disabled={false}
              >
                 Register 
              </Button>
            </Link>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="large"
              bsStyle="primary"
              disabled={true}
              type="submit"
            >
              Login
            </Button>
          </ButtonToolbar>
        </form>,
        "className": "login top-buffer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              bsSize="large"
              controlId="email"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Email
              </ControlLabel>
              <WithInputVaildator
                message="Invalid email"
                pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
                type="email"
              >
                <FormControl
                  autoFocus={true}
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  required={true}
                  type="email"
                  value=""
                />
              </WithInputVaildator>
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              bsSize="large"
              controlId="password"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Password
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="password"
                value=""
              />
            </FormGroup>,
            <ButtonToolbar
              bsClass="btn-toolbar"
              className="pull-right"
            >
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/sign-up",
                  }
                }
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="large"
                  bsStyle="default"
                  disabled={false}
                >
                   Register 
                </Button>
              </Link>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="large"
                bsStyle="primary"
                disabled={true}
                type="submit"
              >
                Login
              </Button>
            </ButtonToolbar>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "bsSize": "large",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Email
                </ControlLabel>,
                <WithInputVaildator
                  message="Invalid email"
                  pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
                  type="email"
                >
                  <FormControl
                    autoFocus={true}
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value=""
                  />
                </WithInputVaildator>,
              ],
              "controlId": "email",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Email",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Email",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <FormControl
                    autoFocus={true}
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value=""
                  />,
                  "message": "Invalid email",
                  "pattern": "^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$",
                  "type": "email",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoFocus": true,
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "onChange": [Function],
                    "required": true,
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "bsSize": "large",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Password
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  type="password"
                  value=""
                />,
              ],
              "controlId": "password",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Password",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Password",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "btn-toolbar",
              "children": Array [
                <Link
                  replace={false}
                  to={
                    Object {
                      "pathname": "/sign-up",
                    }
                  }
                >
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsSize="large"
                    bsStyle="default"
                    disabled={false}
                  >
                     Register 
                  </Button>
                </Link>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="large"
                  bsStyle="primary"
                  disabled={true}
                  type="submit"
                >
                  Login
                </Button>,
              ],
              "className": "pull-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsSize="large"
                    bsStyle="default"
                    disabled={false}
                  >
                     Register 
                  </Button>,
                  "replace": false,
                  "to": Object {
                    "pathname": "/sign-up",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsSize": "large",
                    "bsStyle": "default",
                    "children": " Register ",
                    "disabled": false,
                  },
                  "ref": null,
                  "rendered": " Register ",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsSize": "large",
                  "bsStyle": "primary",
                  "children": "Login",
                  "disabled": true,
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
